libnavui-app/src/main/java/com/mapbox/navigation/ui/app/internal/location/LocationComponent.kt

```diff
@@ -1,18 +1,18 @@
-package com.mapbox.navigation.dropin.component.location
+package com.mapbox.navigation.ui.app.internal.location
```

Comment:
In my opinion, components are something we want external users to be able to use. In this case moving `LocationComponent` to `internal` package makes it's counter to the idea. This should live in `libnavui-maps`.

Reply:
The `LocationComponent` still uses `LocationStateController`, and the `LocationStateController` cannot move to `libnavui-maps`.

I'll move this back to dropin ui for now

Reply:
Was thinking out loud a bit. Also changed mind here, because this may introduce issues if we try to share it with androidauto. Would like to resolve the issues

Just tried updating the location component
```
class LocationComponent(
    private val mapboxMap: MapboxMap,
    private val mapPlugins: MapPluginProviderDelegate,
```

But in order to get the location puck drawable, you need a context. So, we need to also pass the context in with the components
```
DisplayMetrics. A value of 0 means to use the density returned from getConfiguration(). This is equivalent to calling getDrawable(int, Resources.Theme).